go/store/nbs/journal_writer.go

```diff
@@ -94,11 +96,22 @@ func openJournalWriter(ctx context.Context, path string) (wr *journalWriter, exi
 		return nil, true, err
 	}
 
-	return &journalWriter{
+	sendEg, ctx := errgroup.WithContext(ctx)
+	recvEg, ctx := errgroup.WithContext(ctx)
```

Comment:
This seems like a weird lifecycle change for the `ctx` that gets passed here. All of the sudden this `context` needs to be valid for the entire duration of the `journalWriter`, instead of just for the duration of this call...